### ISSUE TITLE
WPF application does not compile anymore moving from Core 3.1 to .NET 5 (assemblies not found in refs folder)

### ISSUE THREAD

**Issue by Skyppid** [dotnet/sdk#14522]
I just tried to upgrade our application to .NET 5. All NuGet packages were updated and our own nuget-referenced projects got bumped to .NET 5 as well (worked fine). All other projects (11 of 12) compile just fine, but the main application using WPF, which references other projects fails with this error:

> 11>Done building project "OurApplication vNext_x4alrbgt_wpftmp.csproj".
> 11>D:\Visual Studio 2019\MSBuild\Current\Bin\Roslyn\Microsoft.CSharp.Core.targets(71,5): warning MSB3052: The parameter to the compiler is invalid, '/define:Customer1EMDEN_(DEBUG)' will be ignored.
> 11>CSC : error CS0006: Metadata file 'E:\Projekte\OurApplication vNext\OurApplication vNext\OurApplication\bin\Customer1\Customer1Location\Debug\Modules\ref\C_Customer1.Customer1Location.dll' could not be found
> 11>CSC : error CS0006: Metadata file 'E:\Projekte\OurApplication vNext\OurApplication vNext\OurApplication\bin\Customer1\Customer1Location\Debug\Modules\ref\C_Customer1.Global.dll' could not be found
> 11>Done building project "OurApplication vNext.csproj" -- FAILED.
> 12>------ Rebuild All started: Project: OurApplication.Tests, Configuration: Debug Any CPU ------
> 12>CSC : error CS0006: Metadata file 'E:\Projekte\OurApplication vNext\OurApplication vNext\OurApplication\bin\Customer1\Customer1Location\Debug\Modules\ref\C_Customer1.Customer1Location.dll' could not be found
> 12>CSC : error CS0006: Metadata file 'E:\Projekte\OurApplication vNext\OurApplication vNext\OurApplication\bin\Customer1\Customer1Location\Debug\Modules\ref\C_Customer1.Global.dll' could not be found
> 12>CSC : error CS0006: Metadata file 'E:\Projekte\OurApplication vNext\OurApplication vNext\OurApplication\bin\Customer1\Customer1Location\Debug\ref\OurApplication.dll' could not be found
> 12>Done building project "OurApplication.Tests.csproj" -- FAILED.
> ========== Rebuild All: 5 succeeded, 2 failed, 5 skipped ==========

The same happens with our NUnit Test project. All are targeting 'net5.0-windows' currently.
Our customer modules are built this way:

```xml
  <PropertyGroup>
    <TargetFramework>net5.0-windows</TargetFramework>
    <Configurations>Debug;Release</Configurations>
    <AssemblyVersion>1.0.390.19623</AssemblyVersion>
    <FileVersion>1.0.390.19623</FileVersion>
    <AssemblyName>C_Customer1.Global</AssemblyName>
    <RootNamespace>OutApplication.Customers.Customer1.Global</RootNamespace>
	  <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
  </PropertyGroup>

  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
    <OutputPath>..\..\..\OutApplication GF\bin\Customer1\Customer1Location\Release\Modules\</OutputPath>
    <DebugType>embedded</DebugType>
    <DebugSymbols>true</DebugSymbols>
  </PropertyGroup>

  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
    <OutputPath>..\..\..\OutApplication GF\bin\Customer1\Customer1Location\Debug\Modules\</OutputPath>
    <DebugType>embedded</DebugType>
    <DebugSymbols>true</DebugSymbols>
  </PropertyGroup>
```

So they're output into the "Modules" folder in our application. In order for the build pipeline to recognize changes only occured in the customer module's project, they're referenced conditionally like this in the main application:

```xml
    <ItemGroup Condition="'$(Configuration)|$(Platform)' == 'Customer1 %28Debug%29|AnyCPU'">
        <ProjectReference Include="..\Customers\Customer1\Customer1.Global\Customer1.Global.csproj">
            <Private>False</Private>
        </ProjectReference>
        <ProjectReference Include="..\Customers\Customer1\Customer1.Customer1Location\Customer1.Customer1Location.csproj">
            <Private>False</Private>
        </ProjectReference>
    </ItemGroup>
```

They're set to private as their references should all be in the main app folder, only the module itself (and important files like .pdb or deps.json remain there).

I cannot figure out what's causing this and why since .NET 5 it expects a "refs" folder and the assemblies being located there. I do not want this, this totally breaks our whole build pipeline.

EDIT: After ages of trial & error .NET 5 is dead for now. I tried it in VS 19 and Rider 2020.3 (EAP) and both spit out totally different errors every time I rebuild. Absolutely random. Either it's the ref folder thing or things like "missing method calls" that simply aren't there. Not sure what is irritating the build tools that much, I assume that there should be basically no difference between 3.1 and 5.

**Comment by Skyppid** [comment]
Meanwhile I believe there's a bug in the toolchain. Probably the issue does not surface right, but as far as I found out using Rider as secondary IDE the nuget packages have an issue:

> 13:09	Install failed (project: **.**, package: MiniLib.Platforms.WPF v3.0.9)
> 			Package restore failed. Rolling back package changes for '**.**'
> 			Package MiniLib.Platforms.WPF 3.0.9 is not compatible with net5.0 (.NETCoreApp,Version=v5.0). Package MiniLib.Platforms.WPF 3.0.9 supports: net5.0-windows7.0 (.NETCoreApp,Version=v5.0)

Although the projects are set to be "net5.0-windows" they compile (or pack) as "net5.0-windows7.0" automatically. And this seems to be unresolvable by the current toolchain.

Furthermore I tried setting them to "net5.0" as those projects all have `<UseWpf>true</UseWpf>` set and following [MSDN](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/overview#available-sdks) this should work as it's detected using that switch. But it is not, it fails compilation saying "windows" has to be specified for WPF projects.

Bummer. Guess I'll have to rollback to Core 3.1 and wait for the port a few weeks.
